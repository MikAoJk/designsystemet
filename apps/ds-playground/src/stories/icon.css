.myIconClassname {
  fill: var(--palette-denim-100);
}
