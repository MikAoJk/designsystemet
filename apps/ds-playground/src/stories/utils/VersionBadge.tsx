import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version = process.env[`@skatteetaten/ds-${packageName}`] ?? '..';
  if (process.env['NODE_ENV'] === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
