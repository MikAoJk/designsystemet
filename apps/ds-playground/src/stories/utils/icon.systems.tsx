import { SystemSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const SystemSVGPaths = findSVGPathElements(SystemSVGPathsAndIcons);
export const SystemIconLabels = makeLabelsFromSVGPathElements(SystemSVGPaths);
