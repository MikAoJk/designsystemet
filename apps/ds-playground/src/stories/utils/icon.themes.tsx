import { ThemeSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const ThemeSVGPaths = findSVGPathElements(ThemeSVGPathsAndIcons);
export const ThemeIconLabels = makeLabelsFromSVGPathElements(ThemeSVGPaths);
