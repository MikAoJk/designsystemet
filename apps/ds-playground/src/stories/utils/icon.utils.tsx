import {
  SystemSVGPathsAndIcons,
  ThemeSVGPathsAndIcons,
} from '@skatteetaten/ds-icons';

export type SVGPathAndIconType =
  | typeof SystemSVGPathsAndIcons
  | typeof ThemeSVGPathsAndIcons;
export type SVGPathAndIconKey = keyof SVGPathAndIconType;

export const makeLabelsFromSVGPathElements = function (
  SVGPathElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, string> {
  const labels: Record<string, string> = {};
  Object.keys(SVGPathElements).forEach((name: string) => {
    labels[name] = name.replace('SVGpath', '');
  });

  return labels;
};

export const findSVGPathElements = function (
  SVGPathAndIconElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, SVGPathAndIconType[SVGPathAndIconKey]> {
  const elements: Record<string, SVGPathAndIconType[SVGPathAndIconKey]> = {};
  Object.keys(SVGPathAndIconElements).forEach((elementName: string) => {
    if (elementName.endsWith('SVGpath')) {
      elements[elementName] =
        SVGPathAndIconElements[elementName as SVGPathAndIconKey];
    }
  });

  return elements;
};
