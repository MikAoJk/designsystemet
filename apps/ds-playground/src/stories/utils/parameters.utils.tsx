import { ArgsTable, Stories, PRIMARY_STORY } from '@storybook/addon-docs';
import type { Parameters } from '@storybook/csf';

type DesignParameters = {
  name: string;
  url: string;
};

export const getParameters = function (
  designParameters: DesignParameters[]
): Parameters {
  return {
    docs: {
      page: (): JSX.Element => {
        return (
          <>
            <Stories title={''} />
            <ArgsTable story={PRIMARY_STORY} />
          </>
        );
      },
    },
    design: designParameters.map((designParameter: DesignParameters) => {
      return {
        name: designParameter.name,
        type: 'figma',
        url: designParameter.url,
      };
    }),
    backgrounds: {
      values: [
        { name: 'Svart', value: '#000' },
        { name: 'Hvit', value: '#fff' },
      ],
    },
  };
};
