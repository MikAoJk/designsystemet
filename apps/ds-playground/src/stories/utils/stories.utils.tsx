export const exampleParameters = {
  docs: {
    source: {
      type: 'code',
      language: 'tsx',
      transform: (source: string): string => {
        return source
          .replace('render: (_args): JSX.Element', 'const Examples = ()')
          .replace(
            'render: function Render(_args): JSX.Element ',
            'const Examples = () => '
          )
          .replace('satisfies Story', '');
      },
    },
  },
  /* _args i en story må finnes slik at props i fanen Controls skjules og er prefikset med underscore for å unngå eslint klaging */
  controls: {
    exclude: /.*/,
  },
};
