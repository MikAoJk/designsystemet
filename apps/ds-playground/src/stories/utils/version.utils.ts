export const getVersion = (
  packageName: string
): {
  major: string;
  minor: string;
  patch: string;
  postfix?: string;
  style?: Record<string, string>;
} => {
  const ver = process.env[`@skatteetaten/${packageName}`] ?? '..';
  const [major, minor, patch] = ver.split('.');
  const majorWithPackageName = packageName ? `${packageName}@${major}` : major;

  const style = {
    color: 'var(--semantic-page-foreground)',
    background: 'var(--semantic-success-background)',
    'font-weight': 'var(--font-weight-bold)',
    'font-size': '13px',
  };

  if (process.env['NODE_ENV'] === 'development') {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: 'develop',
      style: {
        ...style,
        background: 'var(--semantic-danger-background)',
      },
    };
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: process.env['STORYBOOK_BUILD_VERSION'],
      style: {
        ...style,
        background: 'var(--semantic-warning-background)',
      },
    };
  }

  return {
    major: majorWithPackageName,
    minor,
    patch,

    style: {
      ...style,
    },
  };
};
