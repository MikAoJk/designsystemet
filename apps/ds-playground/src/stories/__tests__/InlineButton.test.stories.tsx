import { useState } from 'react';

import { InlineButton } from '@skatteetaten/ds-buttons';
import { positionArr } from '@skatteetaten/ds-core-utils';
import { AddOutlineSVGpath } from '@skatteetaten/ds-icons';
import { expect } from '@storybook/jest';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { userEvent, waitFor, within } from '@storybook/testing-library';

import { wrapper } from './testUtils/storybook.testing.utils';
import { SystemSVGPaths } from '../utils/icon.systems';

const defaultButtonText = 'Legg til rapport';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const inlineButton = canvas.getByRole('button');
    await expect(inlineButton).toBeInTheDocument();
    await expect(inlineButton).toHaveAttribute(attribute, expectedValue);
  };

export default {
  component: InlineButton,
  title: 'Tester/InlineButton',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    iconPosition: {
      table: { disable: true },
      options: [...positionArr],
      control: 'radio',
    },
    svgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    // HTML
    accessKey: { table: { disable: true } },
    disabled: { table: { disable: true } },
    type: { table: { disable: true } },
    // Aria
    ariaDescribedby: { table: { disable: true } },
    // Events
    onBlur: { table: { disable: true } },
    onClick: { table: { disable: true } },
    onFocus: { table: { disable: true } },
  },
} as ComponentMeta<typeof InlineButton>;

const Template: ComponentStory<typeof InlineButton> = (args) => (
  <div data-test-block>
    <InlineButton {...args} svgPath={args.svgPath}>
      {/* eslint-disable-next-line testing-library/no-node-access */}
      {args.children}
    </InlineButton>
  </div>
);

const defaultArgs = {
  children: defaultButtonText,
};

// Når InlineButton har en ref, så får dom button elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLButtonElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ...WithRef.argTypes,
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  imageSnapshot: { disable: true },
};
WithRef.play = verifyAttribute('id', 'dummyIdForwardedFromRef');

// Når InlineButton har en id, så har button-element id
// Når InlineButton har en custom CSS, så vises custom stil
// Når InlineButton har en lang, så har button-element lang
// Når InlineButton har dataTestid, så har button-elementet data-testid satt
export const WithAttributes = Template.bind({});
WithAttributes.storyName = 'With Attributes(FA2-5)';
WithAttributes.args = {
  ...defaultArgs,
  id: 'htmlId',
  className: 'dummyClassname',
  lang: 'nb',
  'data-testid': '123ID',
};
WithAttributes.argTypes = {
  ...WithAttributes.argTypes,
  id: { table: { disable: false } },
  className: { table: { disable: false } },
  lang: { table: { disable: false } },
  'data-testid': { table: { disable: false } },
};
WithAttributes.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const inlineButton = canvas.getByRole('button');
  await expect(inlineButton).toHaveClass('dummyClassname');
  await expect(inlineButton).toHaveAttribute('id', 'htmlId');
  await expect(inlineButton).toHaveAttribute('lang', 'nb');
  await expect(inlineButton).toHaveAttribute('data-testid', '123ID');
};

// Når InlineButton instansieres, får den default iconPosition left
// Knapp må også ha tekst/children
export const Defaults = Template.bind({});
Defaults.storyName = 'Defaults (A1, B1)';
Defaults.args = {
  ...defaultArgs,
};
Defaults.argTypes = {
  ...Defaults.argTypes,
  children: { table: { disable: false } },
};
Defaults.parameters = {
  imageSnapshot: {
    focus: `${wrapper} > button`,
    hover: `${wrapper} > button`,
    click: `${wrapper} > button`,
  },
};
Defaults.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const inlineButton = canvas.getByText(defaultButtonText);
  await expect(inlineButton).toBeInTheDocument();
  await expect(inlineButton).toHaveAttribute('type', 'button');
};

// Når InlineButton har en veldig lang tekst så skal tekst venstrejusteres
export const WithLongText = Template.bind({});
WithLongText.storyName = 'With Long Text (A1)';
WithLongText.args = {
  ...defaultArgs,
  children:
    'Denne knappen har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert. ' +
    'Denne knappen har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert.',
};
WithLongText.argTypes = {
  ...WithLongText.argTypes,
  children: { table: { disable: false } },
};

// Når InlineButton har en veldig lang tekst uten breaking space så skal det brekke over flere linjer
export const WithLongTextAndBreaking = Template.bind({});
WithLongTextAndBreaking.storyName = 'With Long Text And Breaking (A1)';
WithLongTextAndBreaking.args = {
  ...defaultArgs,
  children:
    'Denneknappenharenveldiglangtekst.Sålangatdentvingerframlinjeskift.Nårikkeikonsåskaltekstenværevenstrejusteres.' +
    'Denneknappenharenveldiglangtekst.Sålangatdentvingerframlinjeskift.Nårikkeikonsåskaltekstenværevenstrejusteres.',
};
WithLongTextAndBreaking.argTypes = {
  ...WithLongTextAndBreaking.argTypes,
  children: { table: { disable: false } },
};

// Når InlineButton har ett ikon uten posisjon oppgitt, så vises dette ikonet på venstre side (default).
// Tester også for de icon props som er relevant for saken med systemIcon som er brukt.
export const WithIcon = Template.bind({});
WithIcon.storyName = 'With Icon (A3, A4, B4)';
WithIcon.args = {
  ...defaultArgs,
  svgPath: AddOutlineSVGpath,
};
WithIcon.argTypes = {
  ...WithIcon.argTypes,
  svgPath: { table: { disable: false } },
};
WithIcon.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);

  const inlineButton = canvas.getByRole('button');
  // eslint-disable-next-line testing-library/no-node-access
  const svg = inlineButton.querySelector('svg');
  await expect(svg).toHaveAttribute('viewBox', '0 0 24 24');
  await expect(inlineButton).toBeInTheDocument();
};

// Når InlineButton har et custom ikon, så vises dette ikonet
export const WithCustomIcon = Template.bind({});
WithCustomIcon.storyName = 'With Custom Icon (A3)';
WithCustomIcon.args = {
  ...defaultArgs,
  svgPath: <path d={'M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2Z'} />,
};
WithCustomIcon.argTypes = {
  ...WithCustomIcon.argTypes,
  svgPath: {
    table: { disable: false },
    control: { type: null },
  },
};

// Når InlineButton har en veldig lang tekst og det er et ikon med position right så skal tekst høyrejusteres
export const WithLongTextAndIcon = Template.bind({});
WithLongTextAndIcon.storyName = 'With Long Text And Icon (A3)';
WithLongTextAndIcon.args = {
  ...defaultArgs,
  svgPath: AddOutlineSVGpath,
  children:
    'Denne knappen har en veldig lang tekst med ikon på høyre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er høyrejustert. ' +
    'Denne knappen har en veldig lang tekst med ikon på høyre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er høyrejustert.',
};
WithLongTextAndIcon.argTypes = {
  ...WithLongTextAndIcon.argTypes,
  children: { table: { disable: false } },
  svgPath: { table: { disable: false } },
};

// Når InlineButton har ett ikon med posisjon right, så vises dette ikonet på høyre side.
export const WithIconRight = Template.bind({});
WithIconRight.storyName = 'With Icon Right (A5)';
WithIconRight.args = {
  ...defaultArgs,
  svgPath: AddOutlineSVGpath,
  iconPosition: 'right',
};
WithIconRight.argTypes = {
  ...WithIconRight.argTypes,
  iconPosition: {
    table: { disable: false },
  },
};

// Når InlineButton har prop disabled, så er knapp disabled og stil er satt til disabled
export const WithDisabled = Template.bind({});
WithDisabled.storyName = 'With Disabled (B5)';
WithDisabled.args = {
  ...defaultArgs,
  disabled: true,
};
WithDisabled.argTypes = {
  ...WithDisabled.argTypes,
  disabled: { table: { disable: false } },
};
WithDisabled.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  expect(canvas.getByText(defaultButtonText)).toBeDisabled();
};

// Når InlineButton har prop disabled og ikon er satt, så vises ikonet og knapp er disabled og stil er satt til disabled
export const WithDisabledAndIcon = Template.bind({});
WithDisabledAndIcon.storyName = 'With Disabled And Icon (B5)';
WithDisabledAndIcon.args = {
  ...defaultArgs,
  svgPath: AddOutlineSVGpath,
  disabled: true,
};
WithDisabledAndIcon.argTypes = {
  ...WithDisabledAndIcon.argTypes,
  disabled: { table: { disable: false } },
  svgPath: { table: { disable: false } },
};

// Når InlineButton har prop type, så har button-elementet type satt
export const WithType = Template.bind({});
WithType.storyName = 'With Type (B1)';
WithType.args = {
  ...defaultArgs,
  type: 'submit',
};
WithType.argTypes = {
  ...WithType.argTypes,
  type: { table: { disable: false } },
};
WithType.parameters = {
  imageSnapshot: { disable: true },
};
WithType.play = verifyAttribute('type', 'submit');

// Når InlineButton har aria-describedby, så har button-element aria-describedby
export const WithAriaDescribedby = Template.bind({});
WithAriaDescribedby.storyName = 'With AriaDescribedby (B2)';
WithAriaDescribedby.args = {
  ...defaultArgs,
  ariaDescribedby: 'testid1234',
};
WithAriaDescribedby.argTypes = {
  ...WithAriaDescribedby.argTypes,
  ariaDescribedby: { table: { disable: false } },
};
WithAriaDescribedby.parameters = {
  imageSnapshot: { disable: true },
};
WithAriaDescribedby.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const inlineButton = canvas.getByRole('button');
  await expect(inlineButton).toBeInTheDocument();
  await expect(inlineButton).toHaveAttribute('aria-describedby', 'testid1234');
};

// Når InlineButton har satt accessKey, så har accessKey en verdi
export const WithAccesskey = Template.bind({});
WithAccesskey.storyName = 'With Accesskey (B3)';
WithAccesskey.args = {
  ...defaultArgs,
  accessKey: 'j',
};
WithAccesskey.argTypes = {
  ...WithAccesskey.argTypes,
  accessKey: { table: { disable: false } },
};
WithAccesskey.parameters = {
  imageSnapshot: { disable: true },
};
WithAccesskey.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const inlineButton = canvas.getByRole('button');
  await expect(inlineButton).toBeInTheDocument();
  await expect(inlineButton).toHaveAttribute('accessKey', 'j');
};

// Når brukeren setter focus, blurrer, eller klikker på knappen, så kalles riktig eventHandler
// Eventhandlere endrer tesksten på knappen
const EventHandlersTemplate: ComponentStory<typeof InlineButton> = (args) => {
  const [buttonText, setButtonText] = useState('bruk knapp for å events');
  return (
    <div data-test-block>
      <InlineButton
        {...args}
        onFocus={(event: React.FocusEvent<HTMLButtonElement>): void => {
          setButtonText('Knapp har fått fokus');
          args.onFocus && args.onFocus(event);
        }}
        onBlur={(event: React.FocusEvent<HTMLButtonElement>): void => {
          setButtonText('Knapp har blitt blurret');
          args.onBlur && args.onBlur(event);
        }}
        onClick={(event: React.MouseEvent<HTMLButtonElement>): void => {
          setButtonText('Knapp har blitt klikket på');
          args.onClick && args.onClick(event);
        }}
      >
        {buttonText}
      </InlineButton>
    </div>
  );
};
export const WithEventHandlers = EventHandlersTemplate.bind({});
WithEventHandlers.storyName = 'With EventHandlers (A2 delvis)';
WithEventHandlers.args = {
  ...defaultArgs,
};
WithEventHandlers.argTypes = {
  ...WithEventHandlers.argTypes,
};
WithEventHandlers.parameters = {
  imageSnapshot: { disable: true },
};
WithEventHandlers.play = async ({ args, canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const inlineButton = canvas.getByText('bruk knapp for å events');
  await expect(inlineButton).toBeInTheDocument();
  await inlineButton.focus();
  await waitFor(() => expect(args.onFocus).toHaveBeenCalled());
  await userEvent.tab();
  await waitFor(() => expect(args.onBlur).toHaveBeenCalled());
  await userEvent.click(inlineButton);
  await waitFor(() => expect(args.onClick).toHaveBeenCalled());
};
