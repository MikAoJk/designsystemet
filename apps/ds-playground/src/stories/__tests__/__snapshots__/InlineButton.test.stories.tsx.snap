// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/InlineButton Defaults play-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg   "
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithAccesskey play-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg   "
          accesskey="j"
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithAriaDescribedby play-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg   "
          type="button"
          aria-describedby="testid1234"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithAttributes play-test 1`] = `
<div data-test-block="true">
  <button id="htmlId"
          class="InlineButton_button__2aGtg   dummyClassname"
          lang="nb"
          data-testid="123ID"
          form="formid123"
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithCustomIcon smoke-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg InlineButton_button_withIconLeft__wsxa+  "
          type="button"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2Z">
      </path>
    </svg>
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithDisabled play-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg   "
          disabled
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithDisabledAndIcon smoke-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg InlineButton_button_withIconLeft__wsxa+  "
          disabled
          type="button"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithEventHandlers play-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg   "
          type="button"
  >
    <span class>
      Knapp har blitt klikket på
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithIcon play-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg InlineButton_button_withIconLeft__wsxa+  "
          type="button"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithIconRight smoke-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg  InlineButton_button_withIconRight__FUld4 "
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/InlineButton WithLongText smoke-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg   "
          type="button"
  >
    <span class>
      Denne knappen har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert. Denne knappen har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert.
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithLongTextAndIcon smoke-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg  InlineButton_button_withIconRight__FUld4 "
          type="button"
  >
    <span class>
      Denne knappen har en veldig lang tekst med ikon på høyre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er høyrejustert. Denne knappen har en veldig lang tekst med ikon på høyre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er høyrejustert.
    </span>
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/InlineButton WithRef play-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg   "
          type="button"
          id="dummyIdForwardedFromRef"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithSpinner smoke-test 1`] = `
<div data-test-block="true">
  <div class="flex bottomSpacingXL">
    <button class="InlineButton_button__2aGtg   marginRightM"
            type="button"
    >
      <span class>
        Legg til rapport
      </span>
    </button>
    <button class="InlineButton_button__2aGtg InlineButton_button_withIconLeft__wsxa+  marginRightM"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
        </path>
      </svg>
      <span class>
        Legg til rapport
      </span>
    </button>
    <button class="InlineButton_button__2aGtg  InlineButton_button_withIconRight__FUld4 "
            type="button"
    >
      <span class>
        Legg til rapport
      </span>
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
        </path>
      </svg>
    </button>
  </div>
  <div class="flex">
    <button class="InlineButton_button__2aGtg   marginRightM"
            type="button"
    >
      <span class="InlineButton_hide__u3ooR">
        Legg til rapport
      </span>
      <div aria-live="polite"
           aria-atomic="true"
           class="Spinner_spinner__RJ9Al Spinner_spinner_blue__khcHU Spinner_spinner_flexColumn__3SCpy Spinner_spinner_small__2mxjt InlineButton_spinner__GesB1 "
      >
        <div class="Spinner_spinnerAnimation__nAt8p ">
        </div>
        <span class="Spinner_spinnerTitle__4A7b4 Spinner_srOnly__eRxJZ Spinner_spinnerTitle_centerText__ud193 ">
          Laster inn
        </span>
      </div>
    </button>
    <button class="InlineButton_button__2aGtg InlineButton_button_withIconLeft__wsxa+  marginRightM"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU InlineButton_hide__u3ooR"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
        </path>
      </svg>
      <span class="InlineButton_hide__u3ooR">
        Legg til rapport
      </span>
      <div aria-live="polite"
           aria-atomic="true"
           class="Spinner_spinner__RJ9Al Spinner_spinner_blue__khcHU Spinner_spinner_flexColumn__3SCpy Spinner_spinner_small__2mxjt InlineButton_spinner__GesB1 "
      >
        <div class="Spinner_spinnerAnimation__nAt8p ">
        </div>
        <span class="Spinner_spinnerTitle__4A7b4 Spinner_srOnly__eRxJZ Spinner_spinnerTitle_centerText__ud193 ">
          Laster inn
        </span>
      </div>
    </button>
    <button class="InlineButton_button__2aGtg  InlineButton_button_withIconRight__FUld4 "
            type="button"
    >
      <span class="InlineButton_hide__u3ooR">
        Legg til rapport
      </span>
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz InlineButton_icon__YMesU InlineButton_hide__u3ooR"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
        </path>
      </svg>
      <div aria-live="polite"
           aria-atomic="true"
           class="Spinner_spinner__RJ9Al Spinner_spinner_blue__khcHU Spinner_spinner_flexColumn__3SCpy Spinner_spinner_small__2mxjt InlineButton_spinner__GesB1 "
      >
        <div class="Spinner_spinnerAnimation__nAt8p ">
        </div>
        <span class="Spinner_spinnerTitle__4A7b4 Spinner_srOnly__eRxJZ Spinner_spinnerTitle_centerText__ud193 ">
          Laster inn
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`Tester/InlineButton WithType play-test 1`] = `
<div data-test-block="true">
  <button class="InlineButton_button__2aGtg   "
          type="submit"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;
